temporal-sdk/src/main/java/io/temporal/internal/sync/DeterministicRunnerImpl.java

```diff
@@ -231,6 +231,7 @@ public void runUntilAllBlocked(long deadlockDetectionTimeout) {
         }
       }
       e.setStackDump(dump.toString());
+      log.warn("Workflow potentially deadlocked", e);
```

Comment:
Wouldn't it lead to logging the same error twice?

Reply:
In theory, in practice it's not. I'll figure out why